### ISSUE TITLE
[Cloud-Masthead v1]: a11y Feature/improvement -  support `Esc` key to close the Dropdown in masthead component

### ISSUE THREAD

**Issue by rohitrajmishra** [carbon-design-system/carbon-for-ibm-dotcom#10174]
### The problem

a11y keyboard accessible Feature/improvement -  support `Esc` key to close the Dropdown in masthead

Docs Microservice uses Carbon masthead component.
User requested below missing support for keyboard accesibility

`Esc key lose its function. We use Esc key to close the drop-down list . but now we have to use tab key +shift key to return to the menu title and use enter key to collapse it`

Related Issue: https://github.ibm.com/Bluemix/Bluemix-doc-framework/issues/4551#issuecomment-53637492

### The solution

support `Esc` key to close the Dropdown in masthead component

### Application/website

Bluemix-doc-framework

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by rohitrajmishra** [comment]
Screenshot for the related `mastehead` component
![bug-a11y-Esc-key](https://user-images.githubusercontent.com/10531035/221860264-3aa37ef4-1874-4707-bc08-16c5f5c4e076.png)

**Comment by proeung** [comment]
@rohitrajmishra Thanks for opening this issue. The `Esc` key feature has been already accounted for in the Masthead/L0 v2 update (see [example](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/masthead-v2/web-components/iframe.html?id=components-masthead--with-v-2-data&viewMode=story)), however, the Cloud Masthead is currently inheriting code from v1. We’ve discussed switching the masthead component on the `/cloud` web segment and Cloud Docs app to adopt v2 updates, however, we haven’t solidified a deployment plan yet.

On the Cloud Docs app side, please ask Jenifer Schlotfeldt to coordinate with Keegan Whitla and me and we can share documentation for the Masthead v2 adoption process.

**Comment by zhanttbj** [comment]
@proeung hi team, could you pls fix this issue maybe by April? thanks
